File: src/liberary/matchplan_lib/matchplan/matchplan.rs
use serde::{Deserialize, Serialize};
use std::fmt;

use crate::liberary::matchplan_lib::{division::division::Division, player::player::Player};


#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MatchPlan {
    pub divisions: Vec<Division>,
    pub players: Vec<Player>,
    pub end_timestamp: u64,
    pub pause_end_timestamp: u64,
    pub season: String
}

impl fmt::Display for MatchPlan {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "MatchPlan:\nDivisions:\n")?;
        for division in &self.divisions {
            write!(f, "{}\n", division)?;
        }
        write!(f, "Players:\n")?;
        for player in &self.players {
            write!(f, "{}\n", player)?;
        }
        Ok(())
    }
}

File: PORC-Front/src/models/matchplan/Matchplan.ts
import type { DivisionModel } from "./DivisionModel"

export interface Matchplan {
    season: string,
    divisions: DivisionModel[],
    start_timestamp: number,
    end_timestamp: number,
    pause_end_timestamp: number
}

export function create_division_map(plan: Matchplan): Map<string, number> {
    const division_map = new Map<string, number>();

    const sorted = [...plan.divisions].sort((a, b) => a.order - b.order);

    sorted.forEach((division, index) => {
        for (const player of division.players) {
            division_map.set(player.id, index);
        }
    });

    return division_map;
}